### ISSUE TITLE
Query in hcdev panics

### ISSUE THREAD

**Issue by philipbeadle** [holochain/holochain-proto#690]
If I run a query like
```
  var personas = query({
    Return: {
      Hashes: true,
      Entries: true
    },
    Constrain: {
      EntryTypes: ["persona"]
    }
  })
```
it works fine on hcdev web but gets the following in hcdev test

Test 'personas.6' t+34ms: You can list all of the Personas stored.
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x44be2ab]

goroutine 1 [running]:
github.com/robertkrimen/otto.catchPanic.func1(0xc4205ce438)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/error.go:248 +0x561
panic(0x4b61720, 0x54b80b0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/robertkrimen/otto.catchPanic.func1(0xc4205cb0a0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/error.go:248 +0x561
panic(0x4b61720, 0x54b80b0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/robertkrimen/otto.toValue(0x4becd80, 0x0, 0xc4200cee70, 0xc4200dee00, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/value.go:320 +0xb1b
github.com/robertkrimen/otto.(*_runtime).toValue(0xc42047c600, 0x4becd80, 0x0, 0x445a055, 0xc4205cb070, 0xc4205cb078)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/runtime.go:624 +0x7c
github.com/robertkrimen/otto.(*_runtime).safeToValue.func1()
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/runtime.go:205 +0x45
github.com/robertkrimen/otto.catchPanic(0xc4205cb0c8, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/error.go:251 +0x63
github.com/robertkrimen/otto.(*_runtime).safeToValue(0xc42047c600, 0x4becd80, 0x0, 0x0, 0x53ee400, 0xc4205d83a0, 0xc4200dea80, 0x324)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/runtime.go:204 +0xab
github.com/robertkrimen/otto.Otto.ToValue(0x0, 0xc42047c600, 0x4becd80, 0x0, 0x0, 0x53ee400, 0xc4205d83a0, 0xc4200dea80, 0x324)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/otto.go:631 +0x3f
github.com/holochain/holochain-proto.NewJSRibosome.func11(0xc4202ace80, 0x1, 0x1, 0x53f1340, 0xc420130030, 0xc42047c600, 0x0, 0x0, 0x5, 0x4c865c0, ...)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/jsribosome.go:930 +0xe5b
github.com/holochain/holochain-proto.makeJSFN.func1(0xc42047c600, 0x0, 0x0, 0x5, 0x4c865c0, 0xc4206ac060, 0xc420575e20, 0x1, 0x1, 0xc42070aa70, ...)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/jsribosome.go:1346 +0x19c
github.com/robertkrimen/otto.(*_object).call(0xc4205014a0, 0x5, 0x4c865c0, 0xc4206ac060, 0xc420575e20, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/type_function.go:201 +0x256
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeCallExpression(0xc42047c600, 0xc4205391a0, 0x0, 0x0, 0x0, 0x0, 0x8a4, 0xc42046c000)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:244 +0x52c
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeExpression(0xc42047c600, 0x53eae40, 0xc4205391a0, 0x0, 0x1, 0xc420575de0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:44 +0x5fd
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeCallExpression(0xc42047c600, 0xc420539170, 0x0, 0x0, 0x0, 0xc4205cc1e0, 0x44555ec, 0x4ba1f80)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:193 +0x14e
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeExpression(0xc42047c600, 0x53eae40, 0xc420539170, 0x160, 0x150, 0xc4205cc008)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:44 +0x5fd
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatement(0xc42047c600, 0x53eb280, 0xc4204f8340, 0x0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:86 +0x2fc
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatementList(0xc42047c600, 0xc4204f8310, 0x1, 0x1, 0xc420665470, 0xc4205cc390, 0x4077691)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:121 +0x97
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatement(0xc42047c600, 0x53ead80, 0xc420132b80, 0x0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:30 +0x64f
github.com/robertkrimen/otto.(*_runtime).cmpl_call_nodeFunction(0xc42047c600, 0xc42044b5c0, 0xc420665380, 0xc42028c880, 0x5, 0x4c865c0, 0xc4206ac060, 0xc420575d00, 0x1, 0x1, ...)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate.go:61 +0x4ce
github.com/robertkrimen/otto.(*_object).call(0xc42044b5c0, 0x5, 0x4c865c0, 0xc4206ac060, 0xc420575d00, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/type_function.go:226 +0x8c9
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeCallExpression(0xc42047c600, 0xc420539860, 0x0, 0x0, 0x0, 0x4013628, 0x30, 0x4bc4d60)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:244 +0x52c
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeExpression(0xc42047c600, 0x53eae40, 0xc420539860, 0xc4206ed630, 0x8, 0xc42012fc00)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:44 +0x5fd
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeVariableExpression(0xc42047c600, 0xc4205398c0, 0xc420539e90, 0x28000402bb16, 0x150)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:454 +0x10f
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatement(0xc42047c600, 0x53eb480, 0xc420133ae0, 0x0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:103 +0xfe
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatementList(0xc42047c600, 0xc42008b630, 0x5, 0x5, 0xc420665230, 0xc4205cd290, 0x4077691)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:121 +0x97
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatement(0xc42047c600, 0x53ead80, 0xc420133ac0, 0x0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:30 +0x64f
github.com/robertkrimen/otto.(*_runtime).cmpl_call_nodeFunction(0xc42047c600, 0xc4200cd3e0, 0xc420665140, 0xc42028cf00, 0x5, 0x4c865c0, 0xc4206ac060, 0x55026c8, 0x0, 0x0, ...)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate.go:61 +0x4ce
github.com/robertkrimen/otto.(*_object).call(0xc4200cd3e0, 0x5, 0x4c865c0, 0xc4206ac060, 0x55026c8, 0x0, 0x0, 0xc420575a00, 0x0, 0x0, ...)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/type_function.go:226 +0x8c9
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeCallExpression(0xc42047c600, 0xc420665080, 0x0, 0x0, 0x0, 0x53f6200, 0xc420665050, 0xc4205cdc68)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:244 +0x52c
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeExpression(0xc42047c600, 0x53eae40, 0xc420665080, 0x0, 0xc4204ea520, 0xc4205cdd48)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:44 +0x5fd
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeDotExpression(0xc42047c600, 0xc4206650b0, 0x0, 0xc42046c000, 0x5e421e8)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:257 +0x5a
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeExpression(0xc42047c600, 0x53eaf40, 0xc4206650b0, 0x0, 0xc4205cdec0, 0xc420665050)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:50 +0x92a
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeBinaryExpression_comparison(0xc42047c600, 0xc4206650e0, 0x4b0c1e0, 0xc42012fd80, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:163 +0x46
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeExpression(0xc42047c600, 0x53ead40, 0xc4206650e0, 0x10, 0x4b835e0, 0xc4204ea610)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_expression.go:35 +0xaa4
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatement(0xc42047c600, 0x53eafc0, 0xc4204ea650, 0x0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:62 +0x102f
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeStatementList(0xc42047c600, 0xc4204ea640, 0x1, 0x1, 0x5e421e8, 0x0, 0xc4202ace00)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate_statement.go:121 +0x97
github.com/robertkrimen/otto.(*_runtime).cmpl_evaluate_nodeProgram(0xc42047c600, 0xc42014d490, 0xc4205ce300, 0x0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/cmpl_evaluate.go:17 +0xf1
github.com/robertkrimen/otto.(*_runtime).cmpl_runOrEval.func1()
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/runtime.go:675 +0x44
github.com/robertkrimen/otto.catchPanic(0xc4205ce4a0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/error.go:251 +0x63
github.com/robertkrimen/otto.(*_runtime).cmpl_runOrEval(0xc42047c600, 0x4b0c1e0, 0xc4204ea500, 0x0, 0x0, 0xc4204ea500, 0x5e9a5d8, 0x0, 0x0, 0xc42046c000, ...)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/runtime.go:674 +0x136
github.com/robertkrimen/otto.(*_runtime).cmpl_run(0xc42047c600, 0x4b0c1e0, 0xc4204ea500, 0x0, 0x0, 0x10, 0x4b0c1e0, 0x4c99001, 0xc4204ea500, 0xc4205759c0)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/runtime.go:687 +0x62
github.com/robertkrimen/otto.Otto.Run(0x0, 0xc42047c600, 0x4b0c1e0, 0xc4204ea500, 0x5, 0xc4205ce608, 0x49f3d15, 0xc4203d2900, 0xc4206d2000)
	/Users/philipbeadle/go/src/github.com/robertkrimen/otto/otto.go:295 +0x55
github.com/holochain/holochain-proto.(*JSRibosome).Run(0xc4206b0020, 0xc420138280, 0x1b, 0x53feea0, 0xc4206b0020, 0xc4206d2000, 0x0)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/jsribosome.go:1357 +0xca
github.com/holochain/holochain-proto/apptest.DoTest(0xc4203d2900, 0xc42013f9f2, 0x8, 0x6, 0x0, 0x0, 0x0, 0xc4201361e0, 0x28, 0xc4201be7f0, ...)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/apptest/apptest.go:509 +0x225f
github.com/holochain/holochain-proto/apptest.DoTests(0xc4203d2900, 0xc42013f9f2, 0x8, 0xc42007f100, 0x7, 0x9, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/apptest/apptest.go:325 +0x48e
github.com/holochain/holochain-proto/apptest.test(0xc4203d2900, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc420044400, 0xc4200444f0, 0x4cc080d, 0x1c)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/apptest/apptest.go:717 +0x9c0
github.com/holochain/holochain-proto/apptest.Test(0xc4203d2900, 0x0, 0x0, 0x0, 0xc4203d2900, 0x0, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/apptest/apptest.go:613 +0x64
main.setupApp.func2(0xc42018e6e0, 0xc420065400, 0xc42018e6e0)
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/cmd/hcdev/hcdev.go:443 +0x1bf
github.com/urfave/cli.HandleAction(0x4b21fe0, 0xc420125530, 0xc42018e6e0, 0x0, 0xc4200654a0)
	/Users/philipbeadle/go/src/github.com/urfave/cli/app.go:501 +0xd2
github.com/urfave/cli.Command.Run(0x4ca7ba8, 0x4, 0x0, 0x0, 0xc42012eb60, 0x1, 0x1, 0x4cce0ef, 0x29, 0x0, ...)
	/Users/philipbeadle/go/src/github.com/urfave/cli/command.go:165 +0x4bb
github.com/urfave/cli.(*App).Run(0xc42018a000, 0xc42000c080, 0x2, 0x2, 0x0, 0x0)
	/Users/philipbeadle/go/src/github.com/urfave/cli/app.go:259 +0x740
main.main()
	/Users/philipbeadle/go/src/github.com/holochain/holochain-proto/cmd/hcdev/hcdev.go:931 +0x4f
Philips-MBP:holo-vault philipbeadle$

**Comment by zippy** [comment]
so far can't reproduce.  Need more instructions, but it looks like a bug in the query code.  If you turn on debugging you should see something like:

`Query Code:` 

followed by javascript, and maybe some error in the javascript that's causing Otto to panic.  See if you can get that error.

**Comment by zippy** [comment]
Confirmed crash when running from: https://github.com/holochain/holo-vault/tree/develop

**Comment by zippy** [comment]
This problem appears to be due to binary data in some of the entries, which got changed/fixed in the #582.  @philipbeadle  please try checking out that branch and doing your tests from there...

**Comment by philipbeadle** [comment]
Yep, this works now just ran it fine in 582-gossip-world-model
